s3_prototype/main.tf

```diff
@@ -65,7 +65,8 @@ data "aws_iam_policy_document" "policy_document" {
 }
 
 resource "aws_s3_bucket_policy" "bucket_policy" {
-  bucket     = aws_s3_bucket.bucket.*.id[0]
+  count      = var.attach_s3_policy ? 1 : 0
```

Comment:
main change: only create the `aws_s3_bucket_policy` if `var.attach_s3_policy` is true